### PR TITLE
Move a few login-related helper classes to the utils library

### PR DESCRIPTION
**Pull request by aforcier**
Moves `ViewUtils` and `WPTextInputLayout` to the utils library, where they can be used by the extracted login UI library (once we release a new artifact).

([WIP branch for the login extraction](https://github.com/wordpress-mobile/WordPress-Android/compare/develop...feature/extract-login-flow?expand=1), for reference.)

There might be some other pieces in the future, but these are the most glaring ones so far (with all fragments except Prologue and Epilogue extracted).

cc @hypest

### CODE REVIEW

**Comment by hypest** [comment]
Hey @aforcier , nice to see this happening!

Quick q, are you planning to move `WPLoginInputRow` next? Should be fairly contained in itself I think. Asking since that's basically the only user of the `WPTextInputLayout` widget, and the reason the latter was created was just to work around a couple of issues of the 1st party `TextInputLayout` widget.

File: WordPress/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -329,7 +329,6 @@
     <dimen name="margin_text_input_layout_baseline">20dp</dimen>
     <dimen name="margin_extra_extra_extra_large">92dp</dimen>
     <dimen name="promo_indicator_bullet_size">4dp</dimen>
-    <dimen name="textinputlayout_baseline_correction">2dp</dimen>
```
Comment (hypest) [inline WordPress/src/main/res/values/dimens.xml]: Can we also move `textinputlayout_correction_padding` and `textinputlayout_correction_margin_right`. Those are part of the same set of "hacks" to fix issues of the TextInputLayout an pretty much are used in tandem. WDYT?

File: libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/widgets/AutoResizeTextView.java [hunk 2]
```diff
@@ -16,7 +17,7 @@
  * See http://stackoverflow.com/a/5535672
  *
  */
-public class AutoResizeTextView extends TextView {
+public class AutoResizeTextView extends AppCompatTextView {
```
Comment (hypest) [inline libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/widgets/AutoResizeTextView.java]: Just wondering, can you share some background about this change? Is it some lint related warning or something else? Thanks!

**Comment by aforcier** [comment]
> are you planning to move WPLoginInputRow next perhaps? Should be fairly contained in itself I think. Asking since that's basically the only user of the WPTextInputLayout widget, and the reason the latter was created was just to work around a couple of issues of the 1st party TextInputLayout widget.

I felt that, since `WPLoginInputRow` relies on a layout using login styles, it made most sense to extract it to the login library, and not to utils. Here it is in the in-progress repo for the login flow: [[link]](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/tree/c51ed0de0b2be32f274b8acea69a2703c2e5651a/WordPressLoginFlow/src/main/java/org/wordpress/android/login/widgets).

(`WPTextInputLayout` is also there, but will be dropped once this is merged and I push a new artifact for the utils lib.)

File: WordPress/src/main/res/values/dimens.xml (hunk 1 above)
Comment (aforcier) [reply WordPress/src/main/res/values/dimens.xml]: Good point, done in 1d6f478.

**Comment by hypest** [comment]
> I felt that, since WPLoginInputRow relies on a layout using login styles, it made most sense to extract it to the login library, and not to utils. Here it is in the in-progress repo for the login flow: [link].

Aaah, I see, makes sense. All good 👍

**Comment by hypest** [comment]
LGTM!

**Pull request merged**